docs/index.rst

```diff
@@ -85,6 +85,7 @@ Resources
   asdf/contributing
   asdf/CODE_OF_CONDUCT.md
   asdf/release_and_support
+  asdf/deprecations
```

Comment:
In addition to this can you add a warning/note thing comment highlighting the deprecations near the top of this file? Maybe right before the `Getting Started` section. This will clearly advertise the changes since they are pretty major.